### PR TITLE
Volume root log access

### LINKED ISSUE #46: Add a way to simulate flush delay in MemoryLogDevice for testing.
The default behavior should be to always keep the `commit_pos` and `flush_pos` of MemoryLogDevice instance in sync, but we should have a mode where the flush position must be manually updated.  This will be useful to test scenarios that verify code is correct even when commit gets ahead of flush, which is very common in practice.

### LINKED ISSUE #47: Expose root LogDevice from Volume.
Sometimes it is helpful to access the LogDevice directly.

### PR DESCRIPTION
**Pull request by tonyastolfi**
Fixes #46 
Fixes #47

### CODE REVIEW

File: src/llfs/volume.hpp [hunk 1]
```diff
@@ -197,6 +197,11 @@ class Volume
   //
   const PageRecycler::Metrics& page_recycler_metrics() const;
 
+  LogDevice& root_log() const noexcept
+  {
+    return *this->root_log_;
```
Comment (xwei2289) [inline src/llfs/volume.hpp:202]: Not sure if 
return *(this->root_log_);
will improve readability.

**Comment by xwei2289** [comment]
Looks good in general.
Although I am not familiar with llfs, so still need to understand manual vs. auto flush

File: src/llfs/volume.hpp (hunk 1 above)
Comment (tonyastolfi) [reply src/llfs/volume.hpp:202]: I can change it, if you think it helps!

**Review by xwei2289 — APPROVED** [review]

**Review by bhaskarbora2 — COMMENTED** [review]
looks good !

**Pull request merged**